utils.groovy

```diff
@@ -315,9 +315,27 @@ def get_registry_repos(pipecfg, stream) {
     return registry_repos
 }
 
+// Determine if the config.yaml has a test_architectures entry for
+// this cloud which is intended to limit the architectures that
+// are tested for the given cloud.
+def cloud_testing_enabled_for_arch(pipecfg, cloud, basearch) {
+    if (pipecfg.clouds."${cloud}"?.test_architectures) {
+        // The list exists. Return true if the arch is in the list.
+        if (basearch in pipecfg.clouds."${cloud}".test_architectures) {
+            return true
+        } else {
+            return false
+        }
+    } else {
+        // If no test_architectures defined for cloud then default
+        // to running on all architectures (no limits).
+        return true
+    }
```

Comment:
This can be:

```groovy
return basearch in pipecfg.clouds."${cloud}"?.test_architectures ?: []
```

Reply:
I'm not quite sure it's equivalent. I think this would return false if `test_architectures` wasn't defined for a cloud, which isn't what we want. I did simplify this a little in the next upload, though.